packages/core/src/codewhisperer/util/authUtil.ts

```diff
@@ -442,9 +446,6 @@ export class AuthUtil {
         if (conn === undefined) {
             return buildFeatureAuthState(AuthStates.disconnected)
         }
-        if (!isSsoConnection(conn)) {
-            throw new ToolkitError(`Connection "${conn.id}" is not a valid type: ${conn.type}`)
-        }
```

Comment:
Instead of deleting this you can add the check
```
if (!isSsoConnection(conn) && !isSageMaker()) { ... }
```

Reply:
but this change is not just for Sagemaker, the new client enables chat for both Sso and Iam connections.

Reply:
> but this change is not just for Sagemaker

But this isn't a supported feature yet in all of the code yet right? That is a big product change to the Q extension. AFAIK there is no way to get IAM credentials outside of sagemaker.

Also, for a change like this I would expect unit or integ test updates/additions if possible.

Reply:
Ack, will adjust it now